docs/contracts/v4/concepts/02-pool-manager-and-pool.mdx

```diff
@@ -0,0 +1,175 @@
+and pricing info, and hook information as inputs, checks various conditions to ensure that the pool is set up correctly,
+and sets initial values for the pool.
+
+After all the checks and setup, it announces that a new pool has been created by triggering an `Initialize` event.
+
+# PoolManager.sol
+https://github.com/Uniswap/v4-core/blob/main/src/PoolManager.sol
+
+```solidity
+/// @inheritdoc IPoolManager
+function initialize(PoolKey memory key, uint160 sqrtPriceX96, bytes calldata hookData)
+external
+override
+returns (int24 tick)
+{
+    if (key.fee.isStaticFeeTooLarge()) revert FeeTooLarge();
+
+    // see TickBitmap.sol for overflow conditions that can arise from tick spacing being too large
+    if (key.tickSpacing > MAX_TICK_SPACING) revert TickSpacingTooLarge();
+    if (key.tickSpacing < MIN_TICK_SPACING) revert TickSpacingTooSmall();
+    if (key.currency0 > key.currency1) revert CurrenciesInitializedOutOfOrder();
+    if (!key.hooks.isValidHookAddress(key.fee)) revert Hooks.HookAddressNotValid(address(key.hooks));
+
+    if (key.hooks.shouldCallBeforeInitialize()) {
+        if (key.hooks.beforeInitialize(msg.sender, key, sqrtPriceX96, hookData) != IHooks.beforeInitialize.selector)
+        {
+            revert Hooks.InvalidHookResponse();
+        }
+    }
+
+    PoolId id = key.toId();
+    uint24 protocolFees = _fetchProtocolFees(key);
+    uint24 hookFees = _fetchHookFees(key);
+    tick = pools[id].initialize(sqrtPriceX96, protocolFees, hookFees);
+
+    if (key.hooks.shouldCallAfterInitialize()) {
+        if (
+            key.hooks.afterInitialize(msg.sender, key, sqrtPriceX96, tick, hookData)
+                != IHooks.afterInitialize.selector
+        ) {
+            revert Hooks.InvalidHookResponse();
+        }
+    }
+
+    emit Initialize(id, key.currency0, key.currency1, key.fee, key.tickSpacing, key.hooks);
+}
+```
+Here's what initialize does step-by-step:
+
+1. **Check the Fee**:
+- If the fee for using the pool is too high, then revert error called `FeeTooLarge`.
+
+2. **Check the Tick Spacing**:
+- "Ticks" are like markers in the pool. There's a space between these markers.
+- If the space between the markers is too large, then revert with error `TickSpacingTooLarge`.
+- If it's too small, then revert with error `TickSpacingTooSmall`.
+
+3. **Ensure the Digital Currencies are in Order**:
+- Keys should be sorted numerically
+- If they are in the wrong order, then revert with error `CurrenciesInitializedOutOfOrder`.
+
+4. **Check the Hooks**:
+
+5. **Set Up the Pool**:
+- Create a unique `id` for the pool.
+- Actually set up the pool with the given starting price and fees.
+
+6. **After Setup Actions**:
+- Some hooks might have actions to do after the pool is set up. If they don't respond correctly, revert with `InvalidHookResponse`.
+
+7. **Announce the Pool**:
+- Finally, announce to everyone that a new pool has been created, sharing all its details.
```

Comment:
I think directly referencing the code blocks under each will be helpful.

Also I dont like how this is formatted on the web app. It's hard to read